### DISCUSSION TITLE
Renault Megane 3 BOSE - RADIO NOT CONFIGURED

### DISCUSSION THREAD

**Discussion by LemurMort** [cedricp/ddt4all#1288] [Q&A] [answered]
Hello all!
I experienced a rather unpleasant problem regarding my radio. 
It all started with modifying an OBD cable so I could enable AA. I enabled it succesfully but the curiosity got the best of me.
Using DDT4ALL autoscan found these ECUs:
![image](https://user-images.githubusercontent.com/131789389/234318712-5acf8bb2-e83f-4ba4-9ac2-0f1e98600ef8.png)

So after enabling the AA I went to radio and saw there's a loudspeaker test. I started it, it worked but could not stop it no matter what I did.
So I thought maybe Configuration reset will stop it since it was a bit late and the noise was getting irritant:
![image](https://user-images.githubusercontent.com/131789389/234319745-8e75d6a8-43ed-4c8a-b00d-6907fbc67a49.png)

Well, it didn't.
I started to panic a bit and the only way to stop the noise was to disconnect the battery.
After reconnecting it, the radio doesn't work. After turning the ignition ON a yellow triangle appears and a message Unconfigured radio.

I tried to reset the Rlink by pressing home button 5 times - nothing. I transferred the last Rlink update from USB to a new SD card and tried to reset it again - nothing.

I inserted VIN and CRC number into radio ECU and still nothing.

So, the radio doesn't work, parking sensors don't work (visual and auditive aid), rear camera doesn't work.

Should I do a reset on MFD and which kind of reset? 

Can anyone help? If you need any screenshots, please let me know.
Thanks in advance.

BTW I think this is my last time I'm programming something :D

**Comment by AndyIreland** [comment]
haha - dont worry dont let it put you off we have all been there well lots (including me anyway and a lot of others) with the dreaded EOL and triangle - I fixed mine at the time at home with this  file on a formatted SD card (not map card)  and fixed it that way ..... but I have a Renault Zoe EV without the BIC module and i think when the reset config is done on a car with a BIC I think it does something to mess up the settings with that so I dont know if it will work on yours . but it wont hurt trying if you like: 
https://we.tl/t-fMfgYXpaE6

did you by any chance use PyREN and back up a 'dump' before making any changes by the way? - maybe you didnt but if you did if you restore the dump file that might get it working too.

**Comment by LemurMort** [comment]
Andy, thank you for the file. I'm downloading it as I type this.
I'll try to repeat the process with that file (as I mentioned before, I tried it with my own update (around 500MB) and even with some similar file found on a russian forum (around 2GB).

Unfortunately, I don't have PyREN nor CLIP. only DDT4ALL. Not having a backup wasn't a smart thing to do at all.

We'll see if this will work.

**Comment by AndyIreland** [comment]
hope it does

I think I had to change the after sales date on mine too:

**Comment by LemurMort** [comment]
Sorry, it didn't work.
Upon reset the TomTom logo appears and then the dreaded triangle and a message.
![IMG_20230425_192705](https://user-images.githubusercontent.com/131789389/234355470-23dc8dd5-7810-4cde-ae3a-e864bcf8f3e6.jpg)

The buttons on radio are illuminated, however. I don't rememember if that was the case before all this hell took place.
![Uploading 16824437082546428252314345444156.jpg…]()

**Comment by AndyIreland** [comment]
how much have you got filled out in the ddt4all mfd settings?

**Comment by LemurMort** [comment]
Nothing, I think those setings are not messed up.
In radio ecu I wrote VIN and CRC, nothing else.

**Comment by AndyIreland** [comment]
have you checked both radio and MFD settings?

**Comment by LemurMort** [comment]
Since I enabled AA before these crappy incident after AA is still enabled so that's regarding MFD. I don't think those settings are affected, just the radio. What I know for sure is that in the radio the speakers are 'not present', other settings except VIN and CRC I don't really know how they should be set. Should I set today's date?
Do you have any idea what should I do?

**Comment by shrlnm** [comment]
@LemurMort  have you seen my message to you in https://github.com/cedricp/ddt4all/issues/717 ?

**Comment by AndyIreland** [comment]
@LemurMort  i would need to look to look at your settings in person for the radio and the MFD and I could do over the internet if you have a windows laptop hooked up to the internet and a Bluetooth or USB ELM327 dongle and see if there is anything I can see odd

**Comment by LemurMort** [comment]
> @LemurMort have you seen my message to you in #717 ?

Sorry, I missed that somehow. Thanks! I found PyREN, installed it and now I'll see if I can connect to the car. 
I'll post screenshots.

**Comment by LemurMort** [comment]
I'm a total newb. I got PyREN184, can't get the version 3 from this page to work.
With that one I did connect to the car.
![Screenshot 2023-04-27 213127](https://user-images.githubusercontent.com/131789389/234971746-2287845e-e6da-4388-a824-77231d456c59.png)

I made DUMP files, if that can help (two by each, I don't knowif there's any difference bethween each):
[1682622012_Radio_2012_NoDAB_v2.0_20151012T115515.txt](https://github.com/cedricp/ddt4all/files/11347250/1682622012_Radio_2012_NoDAB_v2.0_20151012T115515.txt)
[1682621931_Radio_2012_NoDAB_v2.0_20151012T115515.txt](https://github.com/cedricp/ddt4all/files/11347251/1682621931_Radio_2012_NoDAB_v2.0_20151012T115515.txt)

[1682622139_BIC_A2_A3_v24_20180620T133153.txt](https://github.com/cedricp/ddt4all/files/11347253/1682622139_BIC_A2_A3_v24_20180620T133153.txt)
[1682622118_BIC_A2_A3_v24_20180620T133153.txt](https://github.com/cedricp/ddt4all/files/11347254/1682622118_BIC_A2_A3_v24_20180620T133153.txt)


Also, here are screenshots I made with DDT4ALL:
RADIO 12 - ECU CONFIGURATION
![RADIO_12-ECU CONFIG 1_3](https://user-images.githubusercontent.com/131789389/234972049-ab04042d-84f2-4b6a-9a27-d5354d46c969.png)
![RADIO_12-ECU CONFIG 2_3](https://user-images.githubusercontent.com/131789389/234972156-72d8f386-6c5a-4be7-ab31-d136812d514f.png)
![RADIO_12-ECU CONFIG 3_3](https://user-images.githubusercontent.com/131789389/234972189-25bb5725-9da9-4f4b-9506-8154f45670e5.png)

MFD 12 - ECU CONFIGURATION
![MFD_12-ECU CONFIG 1_2](https://user-images.githubusercontent.com/131789389/234972405-4c9b07a3-4321-403a-ab5d-a2ae7a009915.png)
![MFD_12-ECU CONFIG 2_2](https://user-images.githubusercontent.com/131789389/234972425-362ed719-c194-44a1-a1c4-83e1d3c5d4b4.png)

MFD 13 - ECU CONFIGURATION
![MFD_13-ECU CONFIG 1_2](https://user-images.githubusercontent.com/131789389/234972505-c9b91a2e-a584-4b87-a22f-9683b18a4414.png)
![MFD_13-ECU CONFIG 2_2](https://user-images.githubusercontent.com/131789389/234972524-8bb0cc50-6a5b-4511-a55c-e789aab1b1ef.png)

MFD 14 - ECU CONFIGURATION
![MFD_14-ECU CONFIG 1_2](https://user-images.githubusercontent.com/131789389/234972583-9455ca07-636d-4890-b821-9f8b0b8bb44c.png)
![MFD_14-ECU CONFIG 2_2](https://user-images.githubusercontent.com/131789389/234972593-ae5439e4-1b57-4460-8e51-9003ecff31fd.png)

@shrlnm I didn't find the option ACI in radio block. Is this because of the PyREN version or? Unfortunately I couldn't follow those instructions from drive.ru.

**Comment by AndyIreland** [comment]
I dont know if any bearing this has @LemurMort  on anything but you have a CRC of 1BF4 (which is correct) on your MFD ecu settings and a CRC of 3851 in your Radio ECU settings????

**Comment by LemurMort** [comment]
I did notice that yesterday but the crc I put into radio is from app My Renault. 
1BF4 is hex, 3851 is decimal; the two have nothing in common 🤷

**Comment by shrlnm** [comment]
@LemurMort the right CRC for your VIN is 1BF4 but not sure that it does matter.
The ACI option is absent in menu because of version. It is only in pyren3.
Please send me a dump from the COMMUNICATIONS UNIT  too (MFD).

**Comment by LemurMort** [comment]
@shrlnm I made DUMP files by getting into those blocks then into DDT screens, then some .xml file would be listed and then by pressing 'enter' I'd get into it and export DUMP files. However, on this third ECU no .xml files were listed and by pressing 'enter' I'd roll a step back in menu 😞 
I'll make screenshots this afternoon.

**Comment by AndyIreland** [comment]
> @LemurMort the right CRC for your VIN is 1BF4 but not sure that it does matter. The ACI option is absent in menu because of version. It is only in pyren3. Please send me a dump from the COMMUNICATIONS UNIT too (MFD).

whats the ACI for @shrlnm

**Comment by AndyIreland** [comment]
> @shrlnm I made DUMP files by getting into those blocks then into DDT screens, then some .xml file would be listed and then by pressing 'enter' I'd get into it and export DUMP files. However, on this third ECU no .xml files were listed and by pressing 'enter' I'd roll a step back in menu 😞 I'll make screenshots this afternoon.

here is the mfd xml for 5.4 
https://we.tl/t-HbneNBqAZs

- put it in your ecus folder of pyren and when it says what ecu dont choose blank, write the exact name  MFD_v5.4_20190228T163812.xml of the file there and then press enter

**Comment by LemurMort** [comment]
@AndyIreland, may I ask how did you get that file?
Is it a generic file or is it based on my car's VIN?

**Comment by AndyIreland** [comment]
out of my ecus file in my program folder. 
i think i might have originally downloaded the larger ecu files from (possibly up on here) called 'larger ECU Zip
No its not based on your VIN - its a generic xml file from the database

**Comment by shrlnm** [comment]
> whats the ACI for
ACI is a option in ECU menu of pyren3

ECU : 11339  FG7610684.XML

```
1  - DE : Device errors
2  - ID : Identifications
3  - SY : System state
4  - SP : System parameters
5  - LC : System configuration
6  - AC : Executing tests
7  - CF : Changing configuration
8  - VP : VIN programming
9  - RZ : Resets
10 - FAV : Favourite Parameters
11 - PRA : Parameters list
12 - ETA : States list
13 - IDA : Identifications list
14 - DDT : DDT screens
**15 - ACI : Auto Config Info**
Q  - <Up>

```

It's for getting references for generation proper autoconfiguration macro

**Comment by LemurMort** [comment]
Guys, do you think I could do some damage if I load .xml from @AndyIreland ?

Reading explanation from @shrlnm I'm leaning more to direction of ACI. If I get version 3 to work...

**Comment by AndyIreland** [comment]
@LemurMort - the xml is not a dump .txt file or a macro file - it wont cure your problem its just the xml file in the database of the ECU's folder for the DDT screens in PyREN

**Comment by AndyIreland** [comment]
> > whats the ACI for
> > ACI is a option in ECU menu of pyren3
> 
> ECU : 11339 FG7610684.XML
> 
> ```
> 1  - DE : Device errors
> 2  - ID : Identifications
> 3  - SY : System state
> 4  - SP : System parameters
> 5  - LC : System configuration
> 6  - AC : Executing tests
> 7  - CF : Changing configuration
> 8  - VP : VIN programming
> 9  - RZ : Resets
> 10 - FAV : Favourite Parameters
> 11 - PRA : Parameters list
> 12 - ETA : States list
> 13 - IDA : Identifications list
> 14 - DDT : DDT screens
> **15 - ACI : Auto Config Info**
> Q  - <Up>
> ```
> 
> It's for getting references for generation proper autoconfiguration macro

so is the ACI in the actual xml file then or the pyren3 program?  @shrlnm

**Comment by LemurMort** [comment]
OK, I loaded Andy's .xml and here are DUMP files from MFD and also again Radio and BIC (so they're all together in one post). 
[1682711838_MFD_v5.4_20190228T163812.txt](https://github.com/cedricp/ddt4all/files/11356931/1682711838_MFD_v5.4_20190228T163812.txt)
[1682711793_MFD_v5.4_20190228T163812.txt](https://github.com/cedricp/ddt4all/files/11356932/1682711793_MFD_v5.4_20190228T163812.txt)
[1682711780_11217.txt](https://github.com/cedricp/ddt4all/files/11356933/1682711780_11217.txt)

[1682622012_Radio_2012_NoDAB_v2.0_20151012T115515.txt](https://github.com/cedricp/ddt4all/files/11356938/1682622012_Radio_2012_NoDAB_v2.0_20151012T115515.txt)
[1682621931_Radio_2012_NoDAB_v2.0_20151012T115515.txt](https://github.com/cedricp/ddt4all/files/11356939/1682621931_Radio_2012_NoDAB_v2.0_20151012T115515.txt)
[1682621926_11193.txt](https://github.com/cedricp/ddt4all/files/11356940/1682621926_11193.txt)

[1682622139_BIC_A2_A3_v24_20180620T133153.txt](https://github.com/cedricp/ddt4all/files/11356942/1682622139_BIC_A2_A3_v24_20180620T133153.txt)
[1682622118_BIC_A2_A3_v24_20180620T133153.txt](https://github.com/cedricp/ddt4all/files/11356943/1682622118_BIC_A2_A3_v24_20180620T133153.txt)
[1682622114_10726.txt](https://github.com/cedricp/ddt4all/files/11356944/1682622114_10726.txt)

**Comment by shrlnm** [comment]
> so is the ACI in the actual xml file then or the pyren3 program?

@AndyIreland  ACI function read references from the ECU. Then you may use these data for generating proper autoconfiguration macros

**Comment by AndyIreland** [comment]
> > so is the ACI in the actual xml file then or the pyren3 program?
> 
> @AndyIreland ACI function read references from the ECU. Then you may use these data for generating proper autoconfiguration macros

oh right thanks.

You know that .cmd  Macro script - does that work (or is there one to work with ) the rlink1 radio/nav with the EOL triangle 'not configured' problem or does it only work on rlink 2 radio/MFD ?

**Comment by AndyIreland** [comment]
> OK, I loaded Andy's .xml and here are DUMP files from MFD and also again Radio and BIC (so they're all together in one post). [1682711838_MFD_v5.4_20190228T163812.txt](https://github.com/cedricp/ddt4all/files/11356931/1682711838_MFD_v5.4_20190228T163812.txt) [1682711793_MFD_v5.4_20190228T163812.txt](https://github.com/cedricp/ddt4all/files/11356932/1682711793_MFD_v5.4_20190228T163812.txt) [1682711780_11217.txt](https://github.com/cedricp/ddt4all/files/11356933/1682711780_11217.txt)
> 
> [1682622012_Radio_2012_NoDAB_v2.0_20151012T115515.txt](https://github.com/cedricp/ddt4all/files/11356938/1682622012_Radio_2012_NoDAB_v2.0_20151012T115515.txt) [1682621931_Radio_2012_NoDAB_v2.0_20151012T115515.txt](https://github.com/cedricp/ddt4all/files/11356939/1682621931_Radio_2012_NoDAB_v2.0_20151012T115515.txt) [1682621926_11193.txt](https://github.com/cedricp/ddt4all/files/11356940/1682621926_11193.txt)
> 
> [1682622139_BIC_A2_A3_v24_20180620T133153.txt](https://github.com/cedricp/ddt4all/files/11356942/1682622139_BIC_A2_A3_v24_20180620T133153.txt) [1682622118_BIC_A2_A3_v24_20180620T133153.txt](https://github.com/cedricp/ddt4all/files/11356943/1682622118_BIC_A2_A3_v24_20180620T133153.txt) [1682622114_10726.txt](https://github.com/cedricp/ddt4all/files/11356944/1682622114_10726.txt)

errrr ... but they are all dumps / rollback files off your own car where the er radio is all unconfigured with a triangle on your screen!

see what happens is that you do it this way **before** you make any changes on your car's ECU's 
Then you make a change or changes then find you have corrupted your ECU settings and think "not to worry PyREN has made a .txt backup/dump" so then you go into that ECU with Pyren then go into the 'Rollback' setting and select the last .txt dump file that pyren automatically backed up for you (well thats the theory on how it works anyway) 

if your ECU is already corrupt or wrongly configured then let Pyren do a dump .txt file all you are doing is backing up unconfigured/incorrectly configured/corrup ECU settings - so if you rollback you are back to square 1 

you could try when you are in PyREN selecting "prefer to use from xml file" instead of 'Prefer ECU settings' possibly and see if that works. i havent tried it though.

**Comment by LemurMort** [comment]
Thanks for the explanation, @AndyIreland . I kind of knew that but as @shrlnm requested earlier I uploaded it. 

There's a great video on yt by user Andrey Amokk about PyREN, macros and ACI. I believe I could solve my case but I can't get bloody PyREN3 to work 😕

**Comment by shrlnm** [comment]
Hello @LemurMort 
I made macro for you
https://cloud.mail.ru/public/35nz/wCH7aCuSU
https://cloud.mail.ru/public/eAaY/5Hm7tVbWS
You may implement them by old pyren.  Do not forget to enable logging before implementation and check that no errors will be in the output. 

What the problem to run pyren3? For running pyren3 you only need to install python v3.xx first. All others is absolutely the same as with old pyren.

**Comment by shrlnm** [comment]
> You know that .cmd Macro script - does that work (or is there one to work with ) the rlink1 radio/nav with the EOL triangle 'not configured' problem or does it only work on rlink 2 radio/MFD ?

Hello @AndyIreland 
With macro scripts you may repair RLink1//MFD as well as Rlink2 (it has no MFD). 

Actually you may repair almost any ECU by this method,  and you don't have to have dumps history now. Actually dumps may be generated from ren databases according to your vin number 

Have you read my articles?
https://www.drive2.ru/b/643019689764615344/
https://www.drive2.ru/b/643355315688994881/

**Comment by LemurMort** [comment]
@shrlnm thank you! I hope I'll try it today.

I'm not profficient in these things and this in combination with lack of time makes me rather not to make any hastly moves again (like resetting the radio configuration 😃).

It takes me some time to explore how to do something. As you wrote to ONLY install Python 3.xx (seems like a simple step to do), I'd be probably googling and whatnot sporadically for days to find that out.

Thanks once more! 🙂

**Comment by LemurMort** [comment]
Ahhhh, look what time it is (02:09). Where do I start...

So, I removed previous version of PyREN and I followed the yt video, installed python 2.7.11, followed all the steps.
I even made my car's documentation. It's in russian. Any chance to get it in english?
Some problem with ELM but I finally connected to the CAN2 bus.

**Problem no.1**
I used this DDT db:
![Screenshot 2023-04-30 014349](https://user-images.githubusercontent.com/131789389/235329607-3ec44e37-b166-4b89-8da4-f5cb34663faa.png)

Again, both Radio and BIC have their .xml files, MFD doesn't:
![Screenshot 2023-04-30 014210](https://user-images.githubusercontent.com/131789389/235329542-2bcdc69d-186a-4f33-b88d-24d923867b81.png)

I go to folder 'ecus' in DDT db archive so I could paste .xml from @AndyIreland but the file is already there. 
I put Andy's file and it's identical:
![Screenshot 2023-04-30 014248](https://user-images.githubusercontent.com/131789389/235329584-9b15ffdb-862d-4a4c-b16c-89776940b2e4.png)

Still, no MFD ecu is offered in DDT screen.


**Problem no.2**
I copied macros from @shrlnm into corresponding folder:
![image](https://user-images.githubusercontent.com/131789389/235329705-85cae8bc-bb91-4402-b31f-1951c9448cfb.png)

I click on macro button on start screen...
![Screenshot 2023-04-30 022647](https://user-images.githubusercontent.com/131789389/235329752-826c5dae-98a9-4b6e-accd-8f6a8d223680.png)

... And this appears:
![Screenshot 2023-04-30 022655](https://user-images.githubusercontent.com/131789389/235329771-39d2a31c-e4fd-4b62-a299-6f3d757d2814.png)

Do you know what's happening?

**Comment by shrlnm** [comment, marked as answer]
Hello @LemurMort 
Let's begin from the end.  
Our main goal is to run macros I made for you. You did everything almost right but ticked "Data Logging/csv" option. It don't work in macro mode. Just disable it and repeat
![im1](https://user-images.githubusercontent.com/16278412/235336214-f315248f-a49b-473f-928b-1a3453ab34c9.png)
This option is only for pyren mode.

All other problems are related to outdated pyren. Recently I fixed the problem with MFD modules. I don't know why but MFD modules in CLIP database have an address '57' but in DDT '58'.

The pyren3 works only with python 3.xx 
https://www.python.org/ftp/python/3.10.11/python-3.10.11-amd64.exe
during installation enable 'Tcl/Tk support' and allow the installer to edit paths 
the pyren3 is there https://gitlab.com/py_ren/pyren/-/archive/pyren3/pyren-pyren3.zip

For generation documentation in english use the pyren_bot with command **/doc _your-vin_ GB**

**Comment by LemurMort** [comment]
Good morning @shrlnm .
Thank you for your instructions.

I installed python 3.10.11.
Do I have to be connected to ELM to run macros?
**EDIT:** If I have to be connected, is there a way to somehow check the macros before running them?

As for pyren3, I extracted the zip here:
![image](https://user-images.githubusercontent.com/131789389/235340013-f015f936-91f1-476e-aca6-61e251cf936f.png)
**EDIT:** Should I put content of folder **C:\PyREN\pyren-pyren3** and replace existing files of folder **C:\PyREN\pyren**?


As for documentation in pyren_bot i got message: 
![image](https://user-images.githubusercontent.com/131789389/235340043-003556e8-9ce6-44c4-833c-4b32c96fc5ed.png)

**Comment by shrlnm** [comment]
@LemurMort You may try a macro in demo mode. In launcher you may enter "--demo" in bottom "Other options" field and press button "Macro"

move files from inside pyren-pyren3 one folder up. main.py and folder pyren3 should be placed next to MTCSAVE.

Also you may try to use pyren_bot once again now.

**Comment by LemurMort** [comment]
Good morning @shrlnm and thank you!

I received the files from pyren_bot yesterday. Interesting stuff.
I didn't know there are dump files. 
If I understand correctly; I can now restore my system with these dump files OR with macros?
In dump files I received I noticed this difference in MFD versions:
![image](https://user-images.githubusercontent.com/131789389/235424927-1e137dbb-8ea4-4dbd-8a6b-cfc97b77be70.png)
![image](https://user-images.githubusercontent.com/131789389/235425005-0ee40f31-2a3f-4c1d-a6c7-ebff271f3ad8.png)
If I were to restore via dump, would this be a problem?
- - - - - - - - -
Starting macros in demo mode works,
I notice a difference in filenames between macros you posted earlier and the ones I got from bot.
![Screenshot 2023-05-01 090328](https://user-images.githubusercontent.com/131789389/235421239-e8f36e62-53ae-4c22-818f-e5ba3b6d5b8a.png)
can2_10_... was identical, but can2_76_... is different. I don't think this matters.

Inside is all identical except at the end:
![Screenshot 2023-05-01 090923](https://user-images.githubusercontent.com/131789389/235421807-0372aeb0-82d0-4300-9268-654892404213.png)
Should I use the ones **with #** or **without #**?

 - - - - - - - - - - - - - - - - - - - - - - - - 

I'm sorry if I misunderstood you. 
I copied contents of folder **C:\PyREN\pyren-pyren3** to **C:\PyREN** but I still don't have the option to select the version pyren3 in launcher after I pressed 'UPDATE' button:
![image](https://user-images.githubusercontent.com/131789389/235422437-26384ead-185b-4f32-ab41-2f3a92d81f87.png)

I downloaded the last version of pyren3 with the mentioned address fixed, I update pyren but still cannot load MFD .xml :/

**EDIT**: From DDT I can select and save the recommended MFD v5.4 xml file, but when I go back to PyREN and select DDT screens for MFD, still no xml is visible.
 - - - - - - - - - - - - 
And regarding the stuff that scares me:
![image](https://user-images.githubusercontent.com/131789389/235423228-fce2fc39-0bfc-464a-a182-b844140a48db.png)

What are the correct steps I should make?
Can I run any macro first or is there a particular order?
After running the macro will the instructions be written what to do next?

**Comment by shrlnm** [comment]
Hello @LemurMort 
For RL1 and RL2 it is better to use macro not 0000000_ dumps.
MFD_v3.5 was used by renault engineers when they prepared configuration for your car. Now you may use v5.4
Also with numbers in name of macro. One of them are built-in for your car in DB and other is corrected to your actual firmware. ( option 76:xxxxxx of the bot)

1101 in the end of macros is command for reload the box, for not to reload it manually.

For running pyren3 you need to run main.py

peren3.exe main.py

**Comment by LemurMort** [comment]
Here's the end result:
![IMG_20230501_133608](https://user-images.githubusercontent.com/131789389/235480474-71245eb2-6e56-4947-8141-c8e5bc59ed92.jpg)

I got Pyren3 to work, ran macros 76: then 10: and reset after every macro and nothing.
So I ran API for MFD and radio, got another set of macros from bot, uncommented last lines, then ran 10: and then 76: and then I heard a beautiful sound from each speaker for maybe 2min. Then I inserted pin (from Renault app, that's not the crc as I thought before) and it works now as it did before.
Made dump files, then activated AA only. I have it in the menu but can't connect phone to it. I'll try with a newer smartphone. 

Thank you all for your help, good will and patience 🙂
@shrlnm your guide now makes a lot of sense to me. At the beggining I didn't understand anything 😃

**Reply by Lionel-John** [reply]
Hi, would it be possible for you to share how you were able to fix this? 

I am not winning following the thread.

**Reply by matzel82** [reply]
> Hi, would it be possible for you to share how you were able to fix this? 
> 
> 
> 
> I am not winning following the thread. 

Mostly autoconfig @ Renault Garage, that is Working

**Comment by AndyIreland** [comment]
glad you got it sorted @LemurMort  and well done for all the help @shrlnm

**Comment by LemurMort** [comment]
Thanks @AndyIreland, certainly me too.
As for AA, it was the cable. 
Now that works too.
Hey google...
![IMG_20230501_191411](https://user-images.githubusercontent.com/131789389/235500343-cc3a33d0-2064-4799-9113-237ea3911e34.jpg)

**Comment by AndyIreland** [comment]
> Thanks @AndyIreland, certainly me too. As for AA, it was the cable. Now that works too. Hey google... ![IMG_20230501_191411](https://user-images.githubusercontent.com/131789389/235500343-cc3a33d0-2064-4799-9113-237ea3911e34.jpg)

great, I was about to suggest a different cable. Same happened to me as well on my car. once i bought a decent one it worked every time

**Comment by runsren** [comment]
This thread got my Smart Media System working, thanks.
Special thanks for the Macro files.

**Comment by stumbaumr** [comment]
Dear @shrlnm , do you think that this might also work with [my Smart EQ from 04/2019](https://www-smart--emotion-de.translate.goog/forum/thread/3533-mit-ddt4all-rumgespielt-loudspeaker-test-angemacht-und-in-den-panikmodus-verfall/?_x_tr_sl=de&_x_tr_tl=ru&_x_tr_hl=de&_x_tr_pto=wapp)?

**Reply by stumbaumr** [reply]
Just came about this thread again. I was able to fix it using DDT4All and screenshots of the configuration from another user in the smart-emotion forum.
[The thread up there](https://github.com/cedricp/ddt4all/discussions/1288#discussioncomment-11141217) shows the solution.

**Reply by rlinkFORfree** [reply]
> Just came about this thread again. I was able to fix it using DDT4All and screenshots of the configuration from another user in the smart-emotion forum. [The thread up there](https://github.com/cedricp/ddt4all/discussions/1288#discussioncomment-11141217) shows the solution.

Twingo with Smart WME chassis number can be automatically restored with Pyren.
[WME4530421Y025297.zip](https://github.com/user-attachments/files/22483377/WME4530421Y025297.zip)

**Comment by shrlnm** [comment]
Hello @stumbaumr. I not sure that your VIN is in configuration DB. You may try to implement configuration from some other ranault car.

**Comment by Budahany24** [comment]
Hello everyone, about the rlink1 with bose i have renault megane 3 with Bose system and i retrofitted the Rlink1 i need the wiring of the bose system to the Rlink1 
And in coding by ddt4all i enabled premium amplifier and bose so do i miss anything in coding?

**Reply by AndyIreland** [reply]
<img width="606" height="478" alt="image" src="https://github.com/user-attachments/assets/6fe83be4-37ee-4e11-bcda-5a3ab1e916fb" />

https://www.facebook.com/download/503066041682720/tomtom%20rlink1%20technical%20info.pdf

I cannot see any BOSE connection on the cable. maybe the BOSE/Amplifier is on another CAN line?

<img width="1179" height="594" alt="image" src="https://github.com/user-attachments/assets/92f56113-482c-41f4-8447-cdef24ba7902" />







You might need to enable it in radio 2012 ecu as well:
<img width="1041" height="600" alt="image" src="https://github.com/user-attachments/assets/288b30c8-2ac2-4900-b1d0-211482b2c872" />

**Reply by Budahany24** [reply]
I didn’t activate it in radio2012 ecu, i will try and see , thank you

**Reply by AndyIreland** [reply]
@Budahany24  - dont press any of the SOFT RESET buttons in DDT4ALL whatever you do

**Comment by rimkaloop** [comment]
Hi ,
I followed @shrlnm instructions,  but I got strange filenames from the bot   : 
<img width="283" height="80" alt="2026-03-08_20h48_41" src="https://github.com/user-attachments/assets/95152d5f-0c7c-44aa-935b-33c34c4ed417" />


 
<img width="820" height="170" alt="2026-03-08_20h07_06" src="https://github.com/user-attachments/assets/f16108ee-3737-4abb-81f3-f3a6c10f9bfa" />
<img width="835" height="121" alt="2026-03-08_20h06_13" src="https://github.com/user-attachments/assets/82e86e0f-a6c5-4116-b3ec-a17c920558d2" />


Command run :     /vin VF1AGVYF058216798 10:259208044R,259157606R,259157606R;76:283C24202R,283C24202R,281151559R


Zoe 2017  Rlink1

Helps are welcome,  thanks

**Comment by rimkaloop** [comment]
@KarelSvo   It worked perfectly ,  you saved my life !
Thank you very much . Is there anything I can do for you ?

**Reply by Lionel-John** [reply]
Hi, can you post the steps or advice that helped you?

**Reply by rimkaloop** [reply]
Hello,
I was on the "unconfigured radio" state because of Kingbolen K8 tool that fucked the radio module.
I follewed theses step https://www.drive2.ru/b/643019689764615344/   but the macros were not well generated.
@karelSvo  provided me good ones that I used on pyren tool,  et voila !

**Reply by v0rtex66** [reply]
@rimkaloop Hi, i have the same problem on kangoo 2019 rlink1 (done by kingbolen k7). What's the way to get the macros? bot isnt working vin VF1FW52H263303716

**Reply by rimkaloop** [reply]
Hi, scroll up,  a forum member provided it for me.
